### PR TITLE
Fix: Adjust inventory group names in molecule tests to follow Ansible guidelines

### LINKED ISSUE #3270: Test: Address warning in ansible when running molecule
### Issue Summary

Following error message is observed when running molecule tests for `eos_designs_unit_tests`.
Possibly other scenarios as well.

```
[WARNING]: Invalid characters were found in group names but not replaced, use
-vvvv to see details
```

### Which component(s) of AVD impacted

eos_designs

### How do you run AVD ?

Ansible CLI (with virtual-env or native python)

### Steps to reproduce

```shell
molecule converge --scenario-name eos_designs_unit_tests
```

### Relevant log output

```shell
INFO     Running from /workspaces/ansible-avd/ansible_collections/arista/avd : ansible-galaxy collection install -vvv --force /workspaces/ansible-avd/ansible_collections/arista/avd
INFO     Running eos_designs_unit_tests > syntax
[WARNING]: Invalid characters were found in group names but not replaced, use
-vvvv to see details

playbook: /workspaces/ansible-avd/ansible_collections/arista/avd/molecule/eos_designs_unit_tests/converge.yml
INFO     Running eos_designs_unit_tests > create
[WARNING]: Invalid characters were found in group names but not replaced, use
-vvvv to see details
```

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by MaheshGSLAB**
## Change Summary

Addressed group name warning in ansible when running molecule.

## Related Issue(s)

Fixes #3270 

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes
Changed the group name as capital and updated hyphen with underscore.

## How to test
Run molecule

### User Checklist

- [x] Verified the changes by running pre-commit and all molecules.
- [x] Implemented only those commands whose output is seen on  show running-config command.
- [ ] Along with EOS config template, updated the documentation template as well.
- [ ] Add the test-case in molecules.
- [ ] Tested the generated config through eapi or cvp on lab.
- [ ] Added meaningfully variable names.
- [ ] Added description only if they provide extra information apart from the key.
- [x] Do not using helping words in description.
- [ ] Added punctuation to all descriptions.
- [ ] One row per entry (per primary key) for documentation template.
- [ ] If  making any key "deprecated", Make sure to move tests with this to the deprecated_vars molecule and make sure to add a test if we are missing it .
- [ ] Use ` ` for key names in description. eg. `name`.
- [ ] Use plural key names if it has multiple elements like- sample_policies.
- [ ] Try to keep less indented j2 code.
- [ ] Used natural sort for "for loops" in j2 templates (after checking on EOS CLI).
- [ ] If  checking child key avoid checking parent key before it (arista.avd.defined).
- [ ] Do not show passwords and password type in documents. Hide it using 'arista.avd.hide_passwords' filter. Dont add column for passwords  and password type in documents.
- [ ] Order of config should be as per EOS cli.
- [ ] Add exclamatory marks(!) at correct position in templates. It should be same as reflected in run-config.
- [ ] Do not use inline if/else in jinja, since it becomes much harder to read the template.
- [ ] Do all possible unit test for config and python code

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/SFLOW_TESTS_LEAFS.yml [hunk 1]
```diff
@@ -36,7 +36,7 @@ l3leaf:
 
 
   node_groups:
-    - group: sflow-tests-leaf-mlag
+    - group: SFLOW_TESTS_LEAF_MLAG
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/SFLOW_TESTS_LEAFS.yml]: this group should not be changed. Only the file names and the inventory.yml

Comment (MaheshGSLAB) [reply ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/SFLOW_TESTS_LEAFS.yml]: fixed

**Review by ClausHolbechArista — APPROVED** [review]
LGTM. Please check if this is also ok for the examples.

File: ansible_collections/arista/avd/molecule/eos_designs_negative_unit_tests/inventory/hosts.yml [hunk 2]
```diff
@@ -125,7 +125,7 @@ all:
             core-interfaces-p2p-links-as:
 
           children:
-            duplicate-vlans-mlag:
+            DUPLICATE_VLANS_MLAG:
```
Comment (Shivani-gslab) [inline ansible_collections/arista/avd/molecule/eos_designs_negative_unit_tests/inventory/hosts.yml:128]: As we are following the convention of `capital letters and underscores` for group names, we should update all the group names in this file (some are currently in `lower_case with underscores`).
like here - 
https://github.com/aristanetworks/avd/blob/e338ba15abd9a05725b805c8eb09424594f0203c/ansible_collections/arista/avd/molecule/eos_designs_negative_unit_tests/inventory/hosts.yml#L8

cc: @ClausHolbechArista

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/molecule/eos_designs_negative_unit_tests/inventory/hosts.yml:128]: Sure that would be great. Thanks

Comment (MaheshGSLAB) [reply ansible_collections/arista/avd/molecule/eos_designs_negative_unit_tests/inventory/hosts.yml:128]: fixed

**Review by Shivani-gslab — APPROVED** [review]
LGTM!

**Pull request merged**